source/System.cpp

```diff
@@ -825,8 +825,9 @@ double System::Danger() const
 {
 	double danger = 0.;
 	for(const auto &fleet : fleets)
-		if(fleet.Get()->GetGovernment()->IsEnemy())
-			danger += static_cast<double>(fleet.Get()->Strength()) / fleet.Period();
+		if(auto *gov = fleet.Get()->GetGovernment())
+			if(gov->IsEnemy())
```

Comment:
```suggestion
			if(gov && gov->IsEnemy())
```
I guess the check for nullptr needs to be added here.

Reply:
The nulltpr check happens in the previous if alreayd.

Reply:
Does whether or not the pointer is null actually get evaluated to a bool for the if statement there?

Reply:
yup

Reply:
Right, assignment and nullptr check both together in the if-statement....

Reply:
Could consider doing the assignment before the if-statement for readability. Assignments having a return-value is a bit of a lesser-known C/C++ feature.